### ISSUE TITLE
TypesScript Exclude<T, S> definition is created once for whole schema

### ISSUE THREAD

**Issue by leqwasd** [YousefED/typescript-json-schema#269]
export interface IConfig {
        presence: IUserLayerPresenceConfig;
        date: IUserDateFieldsConfig;
    }

    export interface IUserLayerPresenceConfig {
        absent?: string[];
        present?: string[];
        rest: Exclude<keyof IUserLayerPresenceConfig, "rest">;
    }

    export interface IUserDateFieldsConfig {
        date?: string[];
        dateTime?: string[];
        rest: Exclude<keyof IUserDateFieldsConfig, "rest">;
    }

it generates only one type in schema and uses that for both interfaces:

	"rest": {
		"$ref": "#/definitions/Exclude"
	}

	"Exclude": {
		"description": "Exclude from T those types that are assignable to U",
		"enum": [
			"absent",
			"present"
		],
		"type": "string"
	},

They should be different.
I know these are some advanced types, just wanted to submit this bug, so you could improve this awesome thing!

**Comment by domoritz** [comment]
Good catch. That's definitely a bug. Would you mind creating a test case as a PR? Then maybe you can send another PR with a fix. Try to make the test case simple (no recursion).
